src/main/java/land/oras/auth/FileStoreAuthenticationProvider.java

```diff
@@ -0,0 +1,40 @@
+package land.oras.auth;
+
+import land.oras.credentials.FileStore;
+import land.oras.credentials.FileStore.Credential;
+import land.oras.exception.ConfigLoadingException;
+
+
+/**
+ * FileStoreAuthenticationProvider is an implementation of the AuthProvider interface.
+ * It retrieves credentials from a FileStore and generates a Basic Authentication header.
+ */
+public class FileStoreAuthenticationProvider implements AuthProvider {
+
+    private final FileStore fileStore;
+    private final String serverAddress;
+    private final UsernamePasswordProvider usernamePasswordAuthProvider;
+
+    /**
+     * Constructor for FileStoreAuthenticationProvider.
+     *
+     * @param fileStore     The FileStore instance to retrieve credentials from.
+     * @param serverAddress The server address for which to retrieve credentials.
+     */
+    public FileStoreAuthenticationProvider(FileStore fileStore, String serverAddress) throws Exception {
```

Comment:
Maybe we should use `ContainerRef` instead of raw string?

By passing `ContainerRef` the consumer can choose directly the credential to use for a specific container reference

By passing a "String" consumer do not know what is it. Doesn't it need to include the protocol? Can it support an image reference etc..?

Reply:
Sure,  Sending ContainerRef will make sense we can access lots of different properties

Reply:
Done @jonesbusy